src/realm/array.cpp

```diff
@@ -712,7 +665,7 @@ void Array::truncate_and_destroy_children(size_t new_size)
     // If the array is completely cleared, we take the opportunity to
     // drop the width back to zero.
     if (new_size == 0) {
-        m_capacity = calc_item_count(get_capacity_from_header(), 0);
+        m_capacity = calc_item_count(get_capacity_from_hdr(), 0);
```

Comment:
The I believe `Array::get_capacity_from_hdr` should be renamed to something more explicit and differentiating than just removing the vowels. This seems very confusing to me.